src/tools/pub/pubspec.md

```diff
@@ -272,6 +277,38 @@ publish_to: none
 {% endprettify %}
 
 
+### False_leaks
+
+When publishing leak detection on the client will abort publishing if it finds
+potential leaks of secret credentials, API keys and cryptographic keys in files
+to be published.
+While detection of possible secrets relies on advanced heuristics using regular
+expressions and entropy thresholds, the detection is not perfect.
+If the leak detection reports a _false positive_ you can specify a list of
+[git-ignore patterns][git-ignore-format] under `false_leaks` in `pubspec.yaml`.
+
+The leak detection will ignore any files matching a pattern in `false_leaks`.
+The example below ignores the file `lib/src/file_with_hardcoded_api_key.dart`
+and all `.pem` files in the `test/localhost_certificates/` folder. Starting a
+[git-ignore pattern][git-ignore-format] with slash `/` ensures that is it only
+considered relative to the root folder.
+
+{% prettify yaml tag=pre+code %}
+false_leaks:
+ - /lib/src/file_with_hardcoded_api_key.dart
+ - /test/localhost_certificates/*.pem
+{% endprettify %}
```

Comment:
@mit-mit what do you think about introducing the `false_leaks` key in `pubspec.yaml`?

Reply:
That name LGTM. I'll have some tweaks to the text later, but they should wait until the technical details are all settled.

Reply:
Hmm, not sure I really like the name `false_leaks`. It just sounds wrong somehow. Did we have any alternative names? Maybe reverse it and use something like `allowed_secrets`?

Reply:
naming ideas:
 * `false_leaks` (because these are _false positives found by leak detection_)
 * `allowed_secrets`
 * `ignored_leaks`

So could be `(allowed|ignored|false)_(leaks|secrets)`.

I started leaning towards `false_leaks`, because what we're asking you to do is to explicitly state that these files do not _leak any secrets_, as in _false-positive leaks_.

The term `allowed_secrets` or `ignores_secrets` implies that _there are secrets present_. This is not necessarily the case, it may just be a false-positive match. Indeed, once published you can argue that by the nature of the package being public, its contents cannot contain a secret (or at-least _the secret_ is no-longer secret).

I do wonder if there is any value in include this ignore list in the `pubspec.yaml`, it has the downside that it will likely increase API response sizes for package version listings. I'm not too worried about that, because this feature will rarely be used, and we could filter it out of these API responses if we really want to (or ever need to).

Making the list of "ignored false-positive leaks detected" part of the package, whether by having it in `pubspec.yaml` or another file that is included in the package, makes sense to me. Because if a security researcher finds a leaked API key in a published package, a list such as `false_leaks` is a strong indicator that the package author actually intended to publish this API key. Hence, we can reasonably assume that the API has been locked down appropriately.

Reply:
let's go with `false_secrets`